### PR TITLE
Chapter1/homework/5

### PR DESCRIPTION
**Pull request by mihailfox**
I implemented the methods for each array operation.

### CODE REVIEW

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 1]
```diff
@@ -9,7 +9,22 @@ public static class ArrayOperations
         /// <param name="array">Input array in a random order.</param>
         public static void Sort(int[] array)
         {
-            // ToDo: implement.
+            if (array != null && array.Length != 0)
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Remember the techniques to de-nest?
```cs
if(array == null || array.Lenght == 0) return;
for (int i = 0; i <= array.Length - 1; i++)
{
  SingleFunctionCall();
}
```

Comment (Almantask) [reply Src/BootCamp.Chapter1/ArrayOperations.cs]: Much more readable like this as you don't have to keep track of all the things that happened/is happening above.

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 2]
```diff
@@ -29,7 +52,16 @@ public static void Reverse(int[] array)
         /// <returns>A new array with the last element removed. If an array is empty or null, returns input array.</returns>
         public static int[] RemoveLast(int[] array)
         {
-            // ToDo: implement.
+            if (array != null && array.Length != 0)
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Inverse if and return immediatly.

Comment (Almantask) [reply Src/BootCamp.Chapter1/ArrayOperations.cs]: Duplicate code with removal in other places

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 3]
```diff
@@ -39,7 +71,15 @@ public static int[] RemoveLast(int[] array)
         /// <returns>A new array with the first element removed. If an array is empty or null, returns input array.</returns>
         public static int[] RemoveFirst(int[] array)
         {
-            // ToDo: implement.
+            if (array != null && array.Length != 0)
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Duplicate code with removal in other places

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 4]
```diff
@@ -51,8 +91,30 @@ public static int[] RemoveFirst(int[] array)
         /// <returns>A new array with element removed at a given index. If an array is empty or null, returns input array.</returns>
         public static int[] RemoveAt(int[] array, int index)
         {
-            // ToDo: implement.
-            return array;
+            if (array == null)
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Duplicate code with removal in other places.
Ideally, reuse this function when removing first or last by passing a proper index.

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 5]
```diff
@@ -63,8 +125,22 @@ public static int[] RemoveAt(int[] array, int index)
         /// <returns>A new array with element added at a given index. If an array is empty or null, returns new array with number in it.</returns>
         public static int[] InsertFirst(int[] array, int number)
         {
-            // ToDo: implement.
-            return array;
+            if (array == null)
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: De-duplicate code by reusing InsertAt.

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 6]
```diff
@@ -9,7 +9,22 @@ public static class ArrayOperations
...
         {
-            // ToDo: implement.
+            if (array != null && array.Length != 0)
+            {
+                int temp;
+                for (int i = 0; i <= array.Length - 1; i++)
+                {
+                    for (int j = i + 1; j < array.Length; j++)
+                    {
+                        if (array[i] > array[j])
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Codacy Issue found: [Refactor this code to not nest more than 3 control flow statements.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 4
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter1/ArrayOperations.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369&bid=16189784)

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 7]
```diff
@@ -39,8 +66,8 @@ public static int[] RemoveLast(int[] array)
         /// <returns>A new array with the first element removed. If an array is empty or null, returns input array.</returns>
         public static int[] RemoveFirst(int[] array)
         {
-            // ToDo: implement.
-            return array;
+            if (array == null || array.Length == 0) return array;
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369)

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 8]
```diff
@@ -51,8 +78,30 @@ public static int[] RemoveFirst(int[] array)
...
+            else if (index < 0 || index >= array.Length)
+            {
+                return array;
+            }
+            else
+            {
+                var tempArray = new int[array.Length - 1];
+                for (int i = 0; i < tempArray.Length; i++)
+                {
+                    if (i < index)
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Codacy Issue found: [Use the '?:' operator here.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369)

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 9]
```diff
@@ -75,8 +130,14 @@ public static int[] InsertFirst(int[] array, int number)
         /// <returns>A new array with element added in the end of array. If an array is empty or null, returns new array with number in it.</returns>
         public static int[] InsertLast(int[] array, int number)
         {
-            // ToDo: implement.
-            return array;
+            if (array == null)
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Codacy Issue found: [Use the '?:' operator here.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 3
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter1/ArrayOperations.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369&bid=16189784)

File: Src/BootCamp.Chapter1/ArrayOperations.cs (hunk 6 above)
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Codacy Issue found: [Refactor this code to not nest more than 3 control flow statements.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369)

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 10]
```diff
@@ -75,8 +134,14 @@ public static int[] InsertFirst(int[] array, int number)
...
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Codacy Issue found: [Use the '?:' operator here.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369)

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 11]
```diff
@@ -51,8 +82,30 @@ public static int[] RemoveFirst(int[] array)
...
+            else if (index < 0 || index >= array.Length)
+            {
+                return array;
+            }
+            else
+            {
+                var tempArray = new int[array.Length - 1];
+                for (int i = 0; i < tempArray.Length; i++)
+                {
+                    if (i < index)
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Codacy Issue found: [Use the '?:' operator here.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 2
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter1/ArrayOperations.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369&bid=16189784)

File: Src/BootCamp.Chapter1/ArrayOperations.cs (hunk 6 above)
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Codacy Issue found: [Refactor this code to not nest more than 3 control flow statements.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369)

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 12]
```diff
@@ -51,8 +77,30 @@ public static int[] RemoveFirst(int[] array)
...
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Codacy Issue found: [Use the '?:' operator here.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter1/ArrayOperations.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369&bid=16189784)

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 13]
```diff
@@ -9,7 +9,22 @@ public static class ArrayOperations
         /// <param name="array">Input array in a random order.</param>
         public static void Sort(int[] array)
         {
-            // ToDo: implement.
+            //if (array != null && array.Length != 0)
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter1/ArrayOperations.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369&bid=16189784)

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 14]
```diff
@@ -51,8 +79,30 @@ public static int[] RemoveFirst(int[] array)
...
+            else if (index < 0 || index >= array.Length)
+            {
+                return array;
+            }
+            else
+            {
+                var tempArray = new int[array.Length - 1];
+                for (int i = 0; i < tempArray.Length; i++)
+                {
+                    if (i < index)
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Codacy Issue found: [Use the '?:' operator here.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 2
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter1/ArrayOperations.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369&bid=16189784)

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 15]
```diff
@@ -51,8 +80,30 @@ public static int[] RemoveFirst(int[] array)
...
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Codacy Issue found: [Use the '?:' operator here.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369)

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 16]
```diff
@@ -7,9 +7,27 @@ public static class ArrayOperations
...
+        {
+            int temp = array[i];
+            array[i] = array[j];
+            array[j] = temp;
+        }
+
         public static void Sort(int[] array)
         {
-            // ToDo: implement.
+            if (array == null) return;
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369)

**Comment by mihailfox** [comment]
I'm aware that one test is failing.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 3
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter1/ArrayOperations.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369&bid=16189784)

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 17]
```diff
@@ -51,8 +85,26 @@ public static int[] RemoveFirst(int[] array)
...
+            if (array == null || index < 0 || index >= array.Length)
+            {
+                return array;
+            }
+            else
+            {
+                var tempArray = new int[array.Length - 1];
+                for (int i = 0; i < tempArray.Length; i++)
+                {
+                    if (i < index)
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Codacy Issue found: [Use the '?:' operator here.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369)

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 18]
```diff
@@ -29,7 +59,12 @@ public static void Reverse(int[] array)
         /// <returns>A new array with the last element removed. If an array is empty or null, returns input array.</returns>
         public static int[] RemoveLast(int[] array)
         {
-            // ToDo: implement.
+            if (array == null) return array;
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369)

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 19]
```diff
@@ -19,7 +40,16 @@ public static void Sort(int[] array)
         /// <param name="array">Input array in a random order.</param>
         public static void Reverse(int[] array)
         {
-            // ToDo: implement.
+            if (array == null) return;
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 2
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter1/ArrayOperations.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369&bid=16189784)

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 20]
```diff
@@ -63,8 +121,14 @@ public static int[] RemoveAt(int[] array, int index)
         /// <returns>A new array with element added at a given index. If an array is empty or null, returns new array with number in it.</returns>
         public static int[] InsertFirst(int[] array, int number)
         {
-            // ToDo: implement.
-            return array;
+            if (array == null)
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Codacy Issue found: [Use the '?:' operator here.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369)

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 21]
```diff
@@ -51,8 +91,26 @@ public static int[] RemoveFirst(int[] array)
...
+            if (array == null || index < 0 || index >= array.Length)
+            {
+                return array;
+            }
+            else
+            {
+                var tempArray = new int[array.Length - 1];
+                for (int i = 0; i < tempArray.Length; i++)
+                {
+                    if (i < index)
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Codacy Issue found: [Use the '?:' operator here.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369)

**Comment by mihailfox** [comment]
Implementation  of InsertAt method

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter1/ArrayOperations.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369&bid=16189784)

File: Src/BootCamp.Chapter1/ArrayOperations.cs (hunk 21 above)
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Codacy Issue found: [Use the '?:' operator here.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter1/ArrayOperations.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369&bid=16189784)

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 22]
```diff
@@ -7,9 +7,30 @@ public static class ArrayOperations
         /// If array empty or null- don't do anything.
         /// </summary>
         /// <param name="array">Input array in a random order.</param>
+        ///
+        private static void ExchangeData(int[] array, int i, int j)
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: ExchangeData sounds vague.
SwapElements sounds better.

Also your comments don't belong to this function, they belong to sort function.
Lastly, code should read like a book. You don't read a summary and then the title. High-level function (sort) is like a title of a book and lower-level function - step function (ExchangeData) is like a chapter. 
Please move the function down.

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 23]
```diff
@@ -7,9 +7,30 @@ public static class ArrayOperations
...
+        }
+
         public static void Sort(int[] array)
         {
-            // ToDo: implement.
+            if (array == null)
+            {
+                return;
+            }
+            if (array.Length != 0)
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Invert if statement to reduce nesting.

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 24]
```diff
@@ -19,7 +40,19 @@ public static void Sort(int[] array)
         /// <param name="array">Input array in a random order.</param>
         public static void Reverse(int[] array)
         {
-            // ToDo: implement.
+            if (array == null)
+            {
+                return;
+            }
+            if (array.Length != 0)
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Invert if statement to reduce nesting

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 25]
```diff
@@ -39,8 +80,7 @@ public static int[] RemoveLast(int[] array)
         /// <returns>A new array with the first element removed. If an array is empty or null, returns input array.</returns>
         public static int[] RemoveFirst(int[] array)
         {
-            // ToDo: implement.
-            return array;
+            return array == null || array.Length == 0 ? array : RemoveAt(array, 0);
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Beautiful

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 26]
```diff
@@ -51,8 +91,19 @@ public static int[] RemoveFirst(int[] array)
         /// <returns>A new array with element removed at a given index. If an array is empty or null, returns input array.</returns>
         public static int[] RemoveAt(int[] array, int index)
         {
-            // ToDo: implement.
-            return array;
+            if (array == null || index < 0 || index >= array.Length)
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Many conditions, consider using a bool variable to explain what these conditions mean altogether

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 27]
```diff
@@ -51,8 +91,19 @@ public static int[] RemoveFirst(int[] array)
...
+            {
+                return array;
+            }
+            else
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: If you return, else is not needed because the function exits.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter1/ArrayOperations.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369&bid=16189784)

File: Src/BootCamp.Chapter1/ArrayOperations.cs (hunk 25 above)
Comment (mihailfox) [reply Src/BootCamp.Chapter1/ArrayOperations.cs]: Thanks!

File: Src/BootCamp.Chapter1/ArrayOperations.cs (hunk 22 above)
Comment (mihailfox) [reply Src/BootCamp.Chapter1/ArrayOperations.cs]: I modified code per request.
Thank you!

File: Src/BootCamp.Chapter1/ArrayOperations.cs (hunk 23 above)
Comment (mihailfox) [reply Src/BootCamp.Chapter1/ArrayOperations.cs]: I moved condition to first If statement

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter1/ArrayOperations.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369&bid=16189784)

File: Src/BootCamp.Chapter1/ArrayOperations.cs (hunk 24 above)
Comment (mihailfox) [reply Src/BootCamp.Chapter1/ArrayOperations.cs]: I moved condition to first If statement

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter1/ArrayOperations.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369&bid=16189784)

File: Src/BootCamp.Chapter1/ArrayOperations.cs (hunk 27 above)
Comment (mihailfox) [reply Src/BootCamp.Chapter1/ArrayOperations.cs]: Modification done. Thanks!

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter1/ArrayOperations.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369&bid=16189784)

File: Src/BootCamp.Chapter1/ArrayOperations.cs (hunk 26 above)
Comment (mihailfox) [reply Src/BootCamp.Chapter1/ArrayOperations.cs]: I moved the chain of conditions to a new bool variable ArgumentsAreNotValid

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter1/ArrayOperations.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887369&bid=16189784)

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 28]
```diff
@@ -19,7 +38,17 @@ public static void Sort(int[] array)
...
+            if (array == null || array.Length == 0)
+            {
+                return;
+            }
+
+            for (int i = 0; i < array.Length / 2; i++)
+            {
+                int temp = array[i];
+                array[i] = array[array.Length - i - 1];
+                array[array.Length - i - 1] = temp;
```
Comment (ellmaster) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: I guess Swap() would work here, too?

Comment (mihailfox) [reply Src/BootCamp.Chapter1/ArrayOperations.cs]: Thanks for pointing it out, I forgot to replace the code.

`SwapElements(array, i, array.Length - i - 1);`

**Pull request closed without merging**